### DISCUSSION TITLE
What are asymptotic notations?

### DISCUSSION THREAD

**Discussion by ankit-akash** [ankit-akash/DSA#22] [Q&A] [answered]
brief details.

**Comment by AlexMi-Ha** [comment, marked as answer]
Asymptotic Notation is to have a rough estimate on the runtime of an algorithm.
These are the 3 most important ones:
Ω : This is for the lower bound
e.g insertion sort: if the list is already sorted it just iterates through the list once 
-> n steps: lower bound: Ω(n)

θ : This notation is for the "exact" fit (ignoring constant factors). It can be named if the Ω and the O are the same

O : This is the upper bound runtime
e.g. insertion sort: if the list is not sorted it has to go through a maximum of n^2 iterations
-> O(n^2)

With these notations, only the broad "behaviour" of the function scaling with growing input is important
so a runtime of 700n^4 + 14n^3 + 12n + 7 would just be O(n^4)
